packages/teleport/src/services/joinToken/joinToken.ts

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import api from 'teleport/services/api';
+import cfg from 'teleport/config';
+import makeJoinToken from './makeJoinToken';
+import { JoinToken, Roles } from './types';
+
+class JoinTokenService {
+  fetchJoinToken(roles: Roles[]): Promise<JoinToken> {
```

Comment:
You've defined `roles` as an array but I only ever see it being used with a single value. I'm assuming this is just for some future proofing? What do you think about allowing it to be....
```suggestion
  fetchJoinToken(roles: Roles[] | Roles): Promise<JoinToken> {
```

Reply:
Hmmm the main reason I made it an array is that the backend accepts multiple roles. Even if we don't need more than one role now, I think it is better to keep this way in case we want to generate a token with multiple roles in the future.

```diff
-  fetchJoinToken(roles: Roles[]): Promise<JoinToken> {
+  fetchJoinToken(roles: Roles[] | Roles): Promise<JoinToken> {
```

I think this would make things harder to read (we also would have to change the method's body to test `roles`' type), not sure if it is worth it to remove a `[]`

Reply:
Sure no problem, implementers choice 😁